includes/Commerce/Orders.php

```diff
@@ -82,10 +82,86 @@ public function create_local_order( Order $remote_order ) {
+					$matching_wc_order_item = $wc_order_item;
+					break;
+				}
+			}
+
+			if ( empty( $matching_wc_order_item ) ) {
+
+				$wc_product = wc_get_product( $wc_product_id );
+
+				if ( ! $wc_product instanceof \WC_Product ) {
+					throw new SV_WC_Plugin_Exception( "Product with WC ID $wc_product_id not found" );
+				}
+
+				$matching_wc_order_item = new \WC_Order_Item_Product();
+				$local_order->add_item( $matching_wc_order_item );
+			}
+
+			$matching_wc_order_item->set_product_id( $wc_product_id );
+			$matching_wc_order_item->set_quantity( $item['quantity'] );
+			$matching_wc_order_item->set_subtotal( $item['quantity'] * $item['price_per_unit']['amount'] );
+			// TODO: should we use the estimated_tax or the captured_tax on the line below?
+			$matching_wc_order_item->set_subtotal_tax( $item['tax_details']['estimated_tax']['amount'] );
+			$matching_wc_order_item->save();
+		}
+
+		// update information from selected_shipping_option
+		$selected_shipping_option = $remote_order->get_selected_shipping_option();
+
+		$local_order->set_shipping_total( $selected_shipping_option['price']['amount'] );
+		$local_order->set_shipping_tax( $selected_shipping_option['calculated_tax']['amount'] );
+
+		// update information from shipping_address
+		$shipping_address = $remote_order->get_shipping_address();
+
+		$local_order->set_shipping_last_name( $shipping_address['name'] );
+		$local_order->set_shipping_address_1( $shipping_address['street1'] );
+		$local_order->set_shipping_address_2( $shipping_address['street2'] );
+		$local_order->set_shipping_city( $shipping_address['city'] );
+		$local_order->set_shipping_state( $shipping_address['state'] );
+		$local_order->set_shipping_postcode( $shipping_address['postal_code'] );
+		$local_order->set_shipping_country( $shipping_address['country'] );
+
+		// update information from estimated_payment_details
+		$estimated_payment_details = $remote_order->get_estimated_payment_details();
+
+		// TODO: confirm if we should use $estimated_payment_details['subtotal']['items'] for something
+		// TODO: confirm if we should use $estimated_payment_details['subtotal']['shipping'] for something
+		// TODO: confirm if we should use $estimated_payment_details['tax'] for something
+		$local_order->set_total( $estimated_payment_details['total_amount']['amount'] );
+		// TODO: confirm if we should use $estimated_payment_details['tax_remitted'] for something
+
+		// update information from buyer_details
+		$buyer_details = $remote_order->get_buyer_details();
+
+		$local_order->set_billing_last_name( $buyer_details['name'] );
+		$local_order->set_billing_email( $buyer_details['email'] );
+		// TODO: confirm if we should use $buyer_details['email_remarketing_option'] for something
+
+		// update order status
+		if ( Order::STATUS_CREATED === $remote_order->get_status() ) {
+			$local_order->set_status( 'processing' );
+		}
+
```

Comment:
One last thing we need to store is the remote order ID, which will be needed by the `find_local_order()` method

Reply:
Added in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1515/commits/f86144a44cb465ab1c91e6f901da42c78a48eefc